packages/itwinui-css/src/dialog/dialog.scss

```diff
@@ -51,11 +51,48 @@
   }
 }
 
-@mixin iui-dialog-default {
-  .iui-dialog {
-    left: 50%;
+@mixin iui-dialog-position {
+  top: 33%;
+  left: 50%;
+  transform: translate(-50%, -33%);
+
+  &[data-position] {
+    position: fixed;
+    top: auto;
+    left: auto;
+    transform: none;
+  }
+
+  &[data-position='centered'] {
```

Comment:
I think ideally `data-position` would only be used for the corner positioning & would not be used for centering.

**Centered:**
```html
<div class="iui-dialog">
```

**Corners:**
```html
<div class="iui-dialog" data-position="top-left">
```

Reply:
It is difficult to try to center the dialog while not affecting the way it aligns to the corners. I decided it would be easier to just include the center as a position for the component so we can see how it looks.

Reply:
^ I'll keep trying though